### ISSUE TITLE
files still get uploaded even the limits of files and fileSize exceeded

### ISSUE THREAD

**Issue by louiskhenghao** [expressjs/multer#524]
Hi, i have noticed that when i trying to upload files that more than the options i've set and the files still get uploaded, no error has thrown and the process never end. It just stuck and never received any callback.

**Scenario** 

1. multer options limits set to 3 files 
```
limits: {
    files: 3
}
```
2. 5 files are attached and submit
3. Expect to get an error but 2 files has uploaded to the storage (google cloud storage/disk), and the request is stuck and callback never hit

Anyone has experienced this before?
Appreciate if anyone has any clue for this, tqvm

**Comment by jsieler** [comment]
Have you found any solution? Having same problem...

**Comment by cornpops** [comment]
the fileSize issue is a duplicate of this https://github.com/expressjs/multer/pull/447

**Comment by Xgamefactory** [comment]
still no fixes. error fires after upload finished.
guess we should check filesize with other tools before starting multer uplload )

**Comment by amran08** [comment]
I'm also facing this issue :(

**Comment by mbiakov** [comment]
+1

**Comment by HarshithaKP** [comment]
@louiskhenghao 

Is this issue still exist ??
If exists, can you provide the server code which you used. Because when tried with the below code, error is been thrown and none of the files are uploaded to disk. Please find the server code which I wrote to recreate the error.


server.js : 
```
var storage = multer.diskStorage({
        destination: './uploads' })

var upload = multer({ storage : storage,
        limits : {
                files : 3
                }
        });

app.get('/', (req, res) =>{
        res.sendFile(__dirname + '/index.html')
});

app.post('/', upload.array('file',5), (req, res) => {
        console.log('Uploaded!');
        res.send('Done');
});

```

error : 
```
MulterError: Too many files
```

**Comment by LinusU** [comment]
closing for lack of response, happy to reopen if more information surfaces
